src/main/java/com/minecolonies/coremod/generation/defaults/DefaultRecipeProvider.java

```diff
@@ -0,0 +1,517 @@
+        registerHutRecipe1x2(consumer, ModBlocks.blockHutDeliveryman, Items.IRON_BOOTS, "iron");
+        registerHutRecipe1(consumer, ModBlocks.blockHutDyer, Items.RED_DYE);
+        registerHutRecipe1(consumer, ModBlocks.blockHutEnchanter, Items.ENCHANTING_TABLE);
+        registerHutRecipe1(consumer, ModBlocks.blockHutFarmer, Items.WOODEN_HOE);
+        registerHutRecipe1x2(consumer, ModBlocks.blockHutFarmer, Items.STONE_HOE, "stone");
+        registerHutRecipe1(consumer, ModBlocks.blockHutFisherman, Items.FISHING_ROD);
+        registerHutRecipe1(consumer, ModBlocks.blockHutFletcher, Items.STRING);
+        registerHutRecipe1(consumer, ModBlocks.blockHutFlorist, Items.POPPY);
+        registerHutRecipe1(consumer, ModBlocks.blockHutGlassblower, Items.GLASS);
+        registerHutRecipe1x2(consumer, ModBlocks.blockHutGuardTower, Items.BOW, "");
+        registerHutRecipe1(consumer, ModBlocks.blockHutHospital, Items.CARROT);
+        registerHutRecipe1(consumer, ModBlocks.blockHutLibrary, Items.BOOK);
+        registerHutRecipe1(consumer, ModBlocks.blockHutLumberjack, Items.WOODEN_AXE);
+        registerHutRecipe1x2(consumer, ModBlocks.blockHutLumberjack, Items.STONE_AXE, "stone");
+        registerHutRecipe1(consumer, ModBlocks.blockHutMechanic, Items.REDSTONE_BLOCK);
+        registerHutRecipe1(consumer, ModBlocks.blockHutMiner, Items.WOODEN_PICKAXE);
+        registerHutRecipe1x2(consumer, ModBlocks.blockHutMiner, Items.STONE_PICKAXE, "stone");
+        registerHutRecipe1(consumer, ModBlocks.blockHutMysticalSite, Items.DIAMOND);
+        registerHutRecipe1(consumer, ModBlocks.blockHutPlantation, Items.CACTUS);
+        registerHutRecipe1(consumer, ModBlocks.blockHutRabbitHutch, Items.RABBIT);
+        registerHutRecipe3(consumer, ModBlocks.blockHutSawmill, Items.WOODEN_AXE);
+        registerHutRecipe1(consumer, ModBlocks.blockHutShepherd, Items.SHEARS);
+        registerHutRecipe3(consumer, ModBlocks.blockHutSifter, Items.STRING);
+        registerHutRecipe1(consumer, ModBlocks.blockHutSmeltery, Items.IRON_INGOT);
+        registerHutRecipe3(consumer, ModBlocks.blockHutStonemason, Items.STONE_BRICKS);
+        registerHutRecipe1(consumer, ModBlocks.blockHutSwineHerder, Items.PORKCHOP);
+        registerHutRecipe1(consumer, ModBlocks.blockHutTavern, blockDecoBarrel_standing.get());
+        registerHutRecipe1(consumer, ModBlocks.blockHutTownHall, buildTool.get());
+        registerHutRecipe1(consumer, ModBlocks.blockHutWareHouse, Tags.Items.CHESTS);
+
+        ShapedRecipeBuilder.shaped(ModBlocks.blockHutCrusher)
+                .pattern("XTX")
+                .pattern("CBC")
+                .pattern("XXX")
+                .define('X', ItemTags.PLANKS)
+                .define('C', Items.COBBLESTONE)
+                .define('B', Items.IRON_INGOT)
+                .define('T', buildTool.get())
+                .unlockedBy("has_items", InventoryChangeTrigger.Instance.hasItems(buildTool.get(), Items.IRON_INGOT))

```

Comment:
what are these kinda arbitrary cost unlocked by things for? Do we add an advancement for each recipe?

Reply:
Yes.  That's what unlocks the vanilla recipe book recipes.  You can't actually create a vanilla recipe datagen without supplying one.